### DISCUSSION TITLE
Parent data is not accessible in custom data directive

### DISCUSSION THREAD

**Discussion by alexej-d** [alpinejs/alpine#1674] [7. Bugs (archived)]
Hey, one of the latests commits broke the possibility to access parent component data in a data directive's init function. See this codepen and check the console: https://codepen.io/alexej_d/pen/XWRWPeg

Accessing parent data from x-init works fine though…

**Comment by ryangjchandler** [comment]
Might be related to #1669. Please look out for the next patch release and downgrade in the meantime.

**Comment by pomartel** [comment]
I found a workaround for components where I need to access the parent scope at initialization (which is only a few ones). In my `Alpine.data` component, I renamed the `init()` function to `initialize()` so  it doesn't get invoked automatically. Then I add `x-init="initialize()"` to my div. https://codepen.io/pomartel/pen/ExmEWeL

**Reply by alexej-d** [reply]
Yeah that's what I meant with the last sentence 👍
